components/datatable/datatable.ts

```diff
@@ -1065,15 +1065,12 @@ export class DataTable implements AfterViewChecked,AfterViewInit,AfterContentIni
     }
 
     hasFilter() {
-        let empty = true;
-        for(let prop in this.filters) {
-            if(this.filters.hasOwnProperty(prop)) {
-                empty = false;
```

Comment:
To minimize changes, I probably should have simply added 
`if (this.filters[prop].value.length) `
in front of
`empty = false;`
in the original code.